pkg/collector/pdb_collector.go

```diff
@@ -0,0 +1,81 @@
+package collector
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+type PDBInfo struct {
+	Name               string `json:"name"`
+	MinAvailable       string `json:"minavailable"`
+	MaxUnavailable     string `json:"maxunavailable"`
+	DisruptionsAllowed int32  `json:"disruptionsallowed"`
+}
+
+// PDBCollector defines a Pod disruption Budget Collector struct
+type PDBCollector struct {
+	kubeconfig *restclient.Config
+	data       map[string]string
+}
+
+// NewPDBCollector is a constructor
+func NewPDBCollector(config *restclient.Config) *PDBCollector {
+	return &PDBCollector{
+		data:       make(map[string]string),
+		kubeconfig: config,
+	}
+}
+
+func (collector *PDBCollector) GetName() string {
+	return "poddisruptionbudget"
+}
+
+// Collect implements the interface method
+func (collector *PDBCollector) Collect() error {
+	// Creates the clientset
+	clientset, err := kubernetes.NewForConfig(collector.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("getting access to K8S failed: %w", err)
+	}
+
+	namespacesList, err := clientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
```

Comment:
nit: we properly can create a context at the top of this function, then reuse it in the list ns / pdb calls

Reply:
Thank you sounds good, 👍 lemme know how it looks now ☕️☕️